Install of 15.0.4snap1 fails (mysql setup problem)

Running as root:
nextcloud.manual-install adminuser somepassword
the command hangs on:
Waiting for PHP...

Checking the system logs:
Feb 23 19:10:57 nextcloud nextcloud.php-fpm[12802]: Waiting for MySQL...
Feb 23 19:10:57 nextcloud nextcloud.mysql[12779]: Starting MySQL
Feb 23 19:10:57 nextcloud nextcloud.apache[12849]: Waiting for PHP...
Feb 23 19:10:59 nextcloud nextcloud.renew-certs[12794]: Saving debug log to /var/snap/nextcloud/current/certs/certbot/logs/letsencrypt.log
Feb 23 19:10:59 nextcloud nextcloud.renew-certs[12794]: - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Feb 23 19:10:59 nextcloud nextcloud.renew-certs[12794]: No renewals were attempted.
Feb 23 19:10:59 nextcloud nextcloud.renew-certs[12794]: No hooks were run.
Feb 23 19:10:59 nextcloud nextcloud.renew-certs[12794]: - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Feb 23 19:11:02 nextcloud mysqld_safe[14025]: Starting mysqld daemon with databases from /var/snap/nextcloud/11688/mysql
Feb 23 19:11:02 nextcloud mysqld[14027]: /snap/nextcloud/11688/bin/mysqld (mysqld 5.7.22-log) starting as process 14027 ...
Feb 23 19:11:02 nextcloud mysqld[14027]: failed to set datadir to /var/snap/nextcloud/11688/mysql/
Feb 23 19:11:02 nextcloud mysqld[14027]: Aborting
Feb 23 19:11:02 nextcloud mysqld[14027]: Binlog end
Feb 23 19:11:02 nextcloud mysqld[14027]: /snap/nextcloud/11688/bin/mysqld: Shutdown complete
Feb 23 19:11:02 nextcloud mysqld_safe[14033]: mysqld from pid file /tmp/pids/mysql.pid ended
Feb 23 19:11:03 nextcloud nextcloud.mysql[12779]: ...... * The server quit without updating PID file (/tmp/pids/mysql.pid).
Feb 23 19:11:03 nextcloud nextcloud.mysql[12779]: sed: can't read /var/snap/nextcloud/11688/mysql/root.ini: No such file or directory
Feb 23 19:11:03 nextcloud nextcloud.mysql[12779]: Waiting for MySQL...

Seems that mysql setup fails (can't set owner of data directory) and the install process just hangs and waits.

OS:
Ubuntu 18.04.2 LTS

Snap info:
installed: 15.0.4snap1 (11688) 196MB -

Hi @BernieO 
Running with sudo produces the same result.
One note it's in a LXD container.

Feb 24 09:22:01 nextcloud nextcloud.nextcloud-fixer[79039]: Waiting for Apache...
Feb 24 09:22:03 nextcloud nextcloud.apache[78985]: Making sure nextcloud is setup...
Feb 24 09:22:03 nextcloud nextcloud.mysql[78999]: 2019-02-24T09:22:03.702197Z 0 [Warning] The syntax '--log_warnings/-W' is deprecated and will be removed in a future release. Please use '--log_error_verbosity' instead.
Feb 24 09:22:03 nextcloud nextcloud.mysql[78999]: 2019-02-24T09:22:03.708108Z 0 [Warning] TIMESTAMP with implicit DEFAULT value is deprecated. Please use --explicit_defaults_for_timestamp server option (see documentation for more details).
Feb 24 09:22:03 nextcloud nextcloud.mysql[78999]: 2019-02-24T09:22:03.717772Z 0 [Note] Ignoring --secure-file-priv value as server is running with --initialize(-insecure) or --bootstrap.
Feb 24 09:22:03 nextcloud nextcloud.mysql[78999]: 2019-02-24T09:22:03.727624Z 0 [Note] mysqld (mysqld 5.7.22-log) starting as process 79216 ...
Feb 24 09:22:03 nextcloud nextcloud.mysql[78999]: 2019-02-24T09:22:03.844621Z 0 [Note] Creating the data directory /var/snap/nextcloud/11688/mysql/
Feb 24 09:22:03 nextcloud nextcloud.mysql[78999]: 2019-02-24T09:22:03.994866Z 0 [ERROR] Can't change data directory owner to mysql
Feb 24 09:22:03 nextcloud nextcloud.mysql[78999]: 2019-02-24T09:22:03.995150Z 0 [ERROR] Aborting
Feb 24 09:22:04 nextcloud nextcloud.mysql[78999]: 2019-02-24T09:22:04.011140Z 0 [Note] Binlog end
Feb 24 09:22:04 nextcloud nextcloud.mysql[78999]: 2019-02-24T09:22:04.037854Z 0 [Note] mysqld: Shutdown complete
Feb 24 09:22:05 nextcloud nextcloud.php-fpm[79016]: Waiting for MySQL...
Feb 24 09:22:05 nextcloud nextcloud.mysql[78999]: Starting MySQL
Feb 24 09:22:06 nextcloud nextcloud.apache[78985]: Waiting for PHP...
Feb 24 09:22:07 nextcloud nextcloud.renew-certs[79023]: Saving debug log to /var/snap/nextcloud/current/certs/certbot/logs/letsencrypt.log
Feb 24 09:22:07 nextcloud nextcloud.renew-certs[79023]: - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Feb 24 09:22:07 nextcloud nextcloud.renew-certs[79023]: No renewals were attempted.
Feb 24 09:22:07 nextcloud nextcloud.renew-certs[79023]: No hooks were run.
Feb 24 09:22:07 nextcloud nextcloud.renew-certs[79023]: - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Feb 24 09:22:11 nextcloud mysqld_safe[80194]: Starting mysqld daemon with databases from /var/snap/nextcloud/11688/mysql
Feb 24 09:22:11 nextcloud mysqld[80196]: /snap/nextcloud/11688/bin/mysqld (mysqld 5.7.22-log) starting as process 80196 ...
Feb 24 09:22:11 nextcloud mysqld[80196]: failed to set datadir to /var/snap/nextcloud/11688/mysql/
Feb 24 09:22:11 nextcloud mysqld[80196]: Aborting
Feb 24 09:22:11 nextcloud mysqld[80196]: Binlog end
Feb 24 09:22:11 nextcloud mysqld[80196]: /snap/nextcloud/11688/bin/mysqld: Shutdown complete
Feb 24 09:22:11 nextcloud mysqld_safe[80204]: mysqld from pid file /tmp/pids/mysql.pid ended
Feb 24 09:22:12 nextcloud nextcloud.mysql[78999]: ...... * The server quit without updating PID file (/tmp/pids/mysql.pid).
Feb 24 09:22:12 nextcloud nextcloud.mysql[78999]: sed: can't read /var/snap/nextcloud/11688/mysql/root.ini: No such file or directory
Feb 24 09:22:12 nextcloud nextcloud.mysql[78999]: Waiting for MySQL...

For me:

```
# lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 9.11 (stretch)
Release:        9.11
Codename:       stretch
```


```
# snap run --shell nextcloud.mysql -c 'mysqld --print-defaults'
mysqld would have been started with the following arguments:
--user=mysql --pid-file=/var/run/mysqld/mysqld.pid --socket=/var/run/mysqld/mysqld.sock --port=3306 --basedir=/usr --datadir=/var/lib/mysql --tmpdir=/tmp --lc-messages-dir=/usr/share/mysql --skip-external-locking --bind-address=127.0.0.1 --key_buffer_size=16M --max_allowed_packet=16M --thread_stack=192K --thread_cache_size=8 --myisam-reco

```
```

# snap run --shell nextcloud.mysql -c 'mysqld --defaults-file=$SNAP/my.cnf --print-defaults'
mysqld would have been started with the following arguments:
--user=root --max_allowed_packet=100M --secure-file-priv=NULL --skip-networking 

```

I have tried the beta as follows:

```

#  snap remove nextcloud                                                                                                               
nextcloud removed

#  snap saved
Set  Snap       Age    Version      Rev    Size    Notes
6    nextcloud  9.96s  19.0.1snap1  22327   2468B  auto

# snap forget 6
Snapshot #6 forgotten.

# snap install nextcloud --channel=latest/beta/issue-913
nextcloud (beta/issue-913) 19.0.1snap1+git1.f4c9f02 from Nextcloud✓ installed

```

```

# snap start nextcloud
Started.

# snap logs -f  nextcloud                                                                                                              
2020-07-22T11:07:21Z nextcloud.mysql[32451]: 2020-07-22T11:07:21.675196Z 0 [ERROR] Aborting
2020-07-22T11:07:21Z nextcloud.mysql[32451]: Starting MySQL
2020-07-22T11:07:23Z nextcloud.renew-certs[32488]: Saving debug log to /var/snap/nextcloud/current/certs/certbot/logs/letsencrypt.log
2020-07-22T11:07:23Z nextcloud.renew-certs[32488]: - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
2020-07-22T11:07:23Z nextcloud.renew-certs[32488]: No renewals were attempted.
2020-07-22T11:07:23Z nextcloud.renew-certs[32488]: No hooks were run.
2020-07-22T11:07:23Z nextcloud.renew-certs[32488]: - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
2020-07-22T11:07:27Z nextcloud.mysql[32451]: ...... * The server quit without updating PID file (/tmp/pids/mysql.pid).
2020-07-22T11:07:27Z nextcloud.mysql[32451]: sed: can't read /var/snap/nextcloud/22369/mysql/root.ini: No such file or directory
2020-07-22T11:07:27Z nextcloud.mysql[32451]: Waiting for MySQL...

```



```

#  snap run --shell nextcloud.mysql -c 'mysqld --print-defaults'
mysqld would have been started with the following arguments:
--user=mysql --pid-file=/var/run/mysqld/mysqld.pid --socket=/var/run/mysqld/mysqld.sock --port=3306 --basedir=/usr --datadir=/var/lib/mysql --tmpdir=/tmp --lc-messages-dir=/usr/share/mysql --skip-external-locking --bind-address=127.0.0.1 --key_buffer_size=16M --max_allowed_packet=16M --thread_stack=192K --thread_cache_size=8 --myisam-recover=BACKUP --query_cache_limit=1M --query_cache_size=16M --log_error=/var/log/mysql/error.log --expire_logs_days=10 --max_binlog_size=100M 


# snap run --shell nextcloud.mysql -c 'mysqld --defaults-file=$SNAP/my.cnf --print-defaults'
mysqld would have been started with the following arguments:
--user=root --max_allowed_packet=100M --secure-file-priv=NULL --skip-networking 

```

--- a/src/mysql/bin/start_mysql
+++ b/src/mysql/bin/start_mysql
@@ -8,7 +8,7 @@ new_install=false
 
 # Make sure the database is initialized (this is safe to run if already
 # initialized)
-if mysqld --initialize-insecure --basedir="$SNAP" --datadir="$SNAP_DATA/mysql" --lc-messages-dir="$SNAP/share"; then
+if mysqld --defaults-file="$SNAP/my.cnf" --initialize-insecure --basedir="$SNAP" --datadir="$SNAP_DATA/mysql" --lc-messages-dir="$SNAP/share"; then
 	new_install=true
 fi
 

--- a/src/mysql/support-files/mysql.server
+++ b/src/mysql/support-files/mysql.server
@@ -202,7 +202,7 @@ case "$mode" in
     then
       # Give extra arguments to mysqld with the my.cnf file. This script
       # may be overwritten at next upgrade.
-      "$bindir/mysqld_safe" --datadir="$datadir" --pid-file="$mysqld_pid_file_path" --lc-messages-dir="$SNAP/share" --socket="$MYSQL_SOCKET" "$other_args" >/dev/null 2>&1 &
+      "$bindir/mysqld_safe" --defaults-file="$SNAP/my.cnf" --datadir="$datadir" --pid-file="$mysqld_pid_file_path" --lc-messages-dir="$SNAP/share" --socket="$MYSQL_SOCKET" "$other_args" >/dev/null 2>&1 &
       wait_for_pid created "$!" "$mysqld_pid_file_path"; return_value=$?
 
       # Make lock for RedHat / SuSE

